master/buildbot/master.py

```diff
@@ -328,6 +335,7 @@ def reconfigService(self, new_config):
         # reconfigure all the services
         yield config.ReconfigurableServiceMixin.reconfigService(self, new_config)
```

Comment:
If we presume that reconfig fails and raises an exception, `change_service_lock` will never be released and a subsequent reconfig once the config is fixed would not be able to proceed since the `change_service_lock` is already held. Unless I've missed something?
Wouldn't it be safer to add a try-finally to make sure the lock is actually released again?